### ISSUE TITLE
TypeScript error when using `"moduleResolution": "node16"`

### ISSUE THREAD

**Issue by cedx** [bevry/istextorbinary#270]
I've recently switched from `"moduleResolution": "node"` to `"moduleResolution": "node16"` in a project using this library. And I get TypeScript errors:

```shell
node_modules/istextorbinary/edition-es2019-esm/index.js:14:24 - error TS7006: Parameter 'filename' implicitly has an 'any' type.
14 export function isText(filename, buffer) {
                          ~~~~~~~~
node_modules/istextorbinary/edition-es2019-esm/index.js:14:34 - error TS7006: Parameter 'buffer' implicitly has an 'any' type.
14 export function isText(filename, buffer) {
                                    ~~~~~~
node_modules/istextorbinary/edition-es2019-esm/index.js:45:26 - error TS7006: Parameter 'filename' implicitly has an 'any' type.
45 export function isBinary(filename, buffer) {
                            ~~~~~~~~
node_modules/istextorbinary/edition-es2019-esm/index.js:45:36 - error TS7006: Parameter 'buffer' implicitly has an 'any' type.
[...]
Found 16 errors in the same file, starting at: node_modules/istextorbinary/edition-es2019-esm/index.js:14
```

It seems that when using `node16` module resolution, the typings located in the `compiled-types` folder are not loaded anymore by the TypeScript compiler.

Cf. https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing
I guess that when we use an `export` map in a library, we must also add a `types` property to this export map...

Node.js 18.2.0, TypeScript 4.7.3, `istextorbinary` 6.0.0.

**Comment by balupton** [comment]
This change will need to be implemented in github.com/bevry/boundation which scaffolds the meta files

**Comment by balupton** [comment]
Boundation successfully updated this package and didn't encounter this error. Is it still a problem for you?

**Comment by cedx** [comment]
Thanks for your efforts. Unfortunately, I no longer use TypeScript (I replaced it with [Haxe](https://haxe.org)), so I don't know if this problem is still present.

But I suppose it is still here because I don't see the typings declared in the proper section of the `package.json` file.

```json
"types": "./compiled-types/",
"exports": {
  "node": {
    "import": "./edition-es2019-esm/index.js",
    "default": "./index.cjs",
    "require": "./edition-es2022/index.js"
  },
  "browser": {
    "import": "./edition-browsers/index.js"
  }
}
```

`types` should be declared in the `exports` section. Something like:

```json
"exports": {
  "node": {
    "types": "./compiled-types/",
    "import": "./edition-es2019-esm/index.js",
    "default": "./index.cjs",
    "require": "./edition-es2022/index.js"
  },
  "browser": {
    "types": "./compiled-types/",
    "import": "./edition-browsers/index.js"
  }
}
```

**Comment by balupton** [comment]
Ahh okay, now I understand. I'll roll this out in my current batch of work.

**Comment by balupton** [comment]
This should be resolved with v8.2.0
https://github.com/bevry/istextorbinary/commit/7ceccaa53df3f19b275f833f0be3a67b57c07883
